bin/app-icon.js

```diff
@@ -48,43 +45,40 @@ program
       platforms,
       adaptiveIcons,
     } = parameters;
-    imagemagickCli.getVersion()
-      .then((version) => {
-        if (!version) {
-          console.error('  Error: ImageMagick must be installed. Try:');
-          console.error('    brew install imagemagick');
-          process.exit(1);
-        }
-      })
-      .then(() => {
-        //  Check we have the files we need.
-        const operations = [];
-        operations.push(errorIfMissing(icon, `Source file '${icon}' does not exist. Add the file or specify source icon with the '--icon' parameter.`));
-        if (adaptiveIcons) {
-          const checkPath = backgroundIcon || 'icon.background.png';
-          operations.push(errorIfMissing(checkPath, `Background icon file '${checkPath}' does not exist. Add the file or specify background icon with the '--background-icon' parameter.`));
-        }
-        if (adaptiveIcons) {
-          const checkPath = foregroundIcon || 'icon.foreground.png';
-          operations.push(errorIfMissing(checkPath, `Foreground icon file '${checkPath}' does not exist. Add the file or specify foreground icon with the '--foreground-icon' parameter.`));
-        }
-        return Promise.all(operations);
-      })
-      .then(() => {
-        //  Generate some icons.
-        return generate({
-          sourceIcon: icon,
-          backgroundIcon,
-          foregroundIcon,
-          searchRoot: search,
-          platforms,
-          adaptiveIcons,
-        });
-      })
-      .catch((generateErr) => {
-        console.error(chalk.red(`An error occurred generating the icons: ${generateErr.message}`));
-        return process.exit(1);
+
+    const version = await imagemagickCli.getVersion();
+
+    if (!version) {
+      console.error('  Error: ImageMagick must be installed. Try:');
+      console.error('    brew install imagemagick');
+      return process.exit(1);
+    }
+
+    const operations = [];
+    operations.push(errorIfMissing(icon, `Source file '${icon}' does not exist. Add the file or specify source icon with the '--icon' parameter.`));
+    if (adaptiveIcons) {
+      const checkPath = backgroundIcon || 'icon.background.png';
```

Comment:
This is a *huge* smell, there is a whole set of validation/coercion logic (I think in `validate.js` or something), but it happens *after* all of this logic to check the flags, which no longer makes sense. I will raise a separate refactor ticket, but just a heads up that this is currently probably the biggest structural issue with the code.

Reply:
Yep lets raise a seperate ticket for refactoring outside of `async`/`await`

Reply:
#122 raised